cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -307,6 +314,7 @@ public boolean updateAppAllowed(ApplicationId appId) throws Exception {
   public ApplicationWithPrograms updateApp(ApplicationId appId, AppRequest appRequest,
                                            ProgramTerminator programTerminator) throws Exception {
 
+    ensureAccess(appId);
```

Comment:
This is to not expose the existence of the app if the user does not have any privilege on the app/program, the enforcement is happening in line 325: https://github.com/caskdata/cdap/blob/57e6e376698807377e95f2934bc5a4976cc60484/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java#L325